### ISSUE TITLE
Unable to run bin/logstash - no gem found error

### ISSUE THREAD

**Issue by talevy** [elasticsearch/logstash#2400]
On some environments, I am unable to successfully run `bin/logstash` after running `rake bootstrap` with a clean vendor dir. 

here is the current error I encounter: https://gist.github.com/talevy/0cd47c1dfe9c7f918e79

**Comment by suyograo** [comment]
this worked prior to https://github.com/elasticsearch/logstash/commit/3cb1aa8658d669a99c8f4434cce66d3b5afc0ae1

**Comment by purbon** [comment]
I tried to reproduce this very same issue and was not able to, here basically what I did:

* clean environment.
* run rake bootstrap
* run rake test:prep and install the rubydebug plugin (otherwise there are missing plugins so the command will not work)
* run bin/logstash

and everything works.

Sorry could not reproduce it.

**Comment by ph** [comment]
@talevy 
I did:

```
git clone https://github.com/elasticsearch/logstash.git
rake bootstrap
/t/logstash git:master ❯❯❯ bin/logstash -e "{}"                                                                                                                                                                                                                                                                                                                         
Error: Expected one of #, input, filter, output at line 1, column 1 (byte 1) after
You may be interested in the '--configtest' flag which you can
use to validate logstash's configuration before you choose
to restart a running system.
```

**Comment by ph** [comment]
Look like there is an issue when using local ruby (USE_RUBY=1)

```
git clone https://github.com/elasticsearch/logstash.git
USE_RUBY=1 rake bootstrap # will ignore the setting and vendor jruby.
```
After the installation the environment doesnt work..
``` 
t/logstash git:master ❯❯❯ USE_RUBY=1 bin/logstash -e "{}"                                                                                                                                                                                                                                                                                                                         
LoadError: no such file to load -- bundler

**Comment by purbon** [comment]
Did the same thing like you and got this message:

```
The error reported is: 
  Couldn't find any input plugin named 'stdin'. Are you sure this is correct? Trying to load the stdin input plugin resulted in this error: no such file to load -- logstash/inputs/stdin
```

I guess it all related to having bundler previously installed or not.

**Comment by ph** [comment]
@purbon Ty.
@talevy are you using rvm or rbenv?

**Comment by ph** [comment]
the culprit 

```
~/Code/es/talevy-logstash (master)$ cat ~/.bundle/config 
---
BUNDLE_PATH: vendor
BUNDLE_DISABLE_SHARED_GEMS: "1"
```
My guess is `Bundler.setup` will try to find the a .bundle/config in the hierarchy.

**Comment by talevy** [comment]
thanks @ph. Do you know of any more that needs to be done here to fix this. Or is this more of an issue on my side?

If it is a nonissue, I can close this issue

**Comment by ph** [comment]
@talevy we found the issue on @colinsurprenant and we fixed it. A PR should be done in a few minutes.

**Comment by colinsurprenant** [comment]
fix in #2449 
@talevy this should allow you to keep your `~/.bundle/config ` intact

**Comment by talevy** [comment]
@colinsurprenant's fix in https://github.com/elasticsearch/logstash/pull/2449 solves this issue.
